frontend/packages/integration-tests-cypress/tests/i18n/pseudolocalization.spec.ts

```diff
@@ -14,38 +20,134 @@ describe('Localization', () => {
-    cy.log('test utilization card components');
-    cy.visit('/dashboards?pseudolocalization=true&lng=en');
-    cy.byTestID('utilization-card-item-text').isPseudoLocalized();
-  });
+  const testObjs = OrderedMap<string, LocalizationTestProps>()
+    .set('Masthead', {
+      url: '/settings/cluster',
+      testIDs: ['application-launcher-item'],
+    })
+    .set('Sidebar navigation', {
+      url: '/settings/cluster',
+      testIDs: ['nav'],
+    })
+    .set('Cluster dashboard', {
+      url: '/dashboards',
+      testIDs: [
+        'activity',
+        'activity-recent-title',
+        'ongoing-title',
+        'events-view-all-link',
+        'events-pause-button',
+        'utilization-card-item-text',
+      ],
+    })
+    .set('Events', {
+      url: '/k8s/all-namespaces/events',
+      testIDs: ['timestamp'],
+    })
+    .set('Services list', {
+      url: '/k8s/all-namespaces/services',
+      selectors: [tableColumnHeaders], // all table column headers should be translated
+    })
+    .set('Service details', {
+      url: '/k8s/all-namespaces/services/alertmanager-main',
+      // all section headings, detail item labels, and elements with `data-i18n-translated` attribute should be translated
+      selectors: [sectionHeadings, detailLabels, i18nTranslated],
+      testIDs: ['timestamp'],
+    })
+    .set('Routes list', {
+      url: '/k8s/all-namespaces/routes',
+      selectors: [
+        '.pf-c-dropdown__group-title',
+        '.co-filter-dropdown-item__name',
+        tableColumnHeaders,
+      ],
+    })
+    .set('Route details', {
+      url: '/k8s/all-namespaces/routes/alertmanager-main',
+      selectors: [sectionHeadings, detailLabels, i18nTranslated, '.graph-title'],
+      testIDs: ['timestamp'],
+    })
+    .set('Ingresses list', {
+      url: '/k8s/all-namespaces/ingresses',
+      createResource: true,
+      selectors: [tableColumnHeaders],
+    })
+    .set('Ingress details', {
+      url: '/k8s/all-namespaces/ingresses/example',
+      selectors: [sectionHeadings, detailLabels, i18nTranslated],
+      testIDs: ['timestamp'],
+      deleteResource: 'Ingress',
+    })
+    .set('Policies list', {
+      url: '/k8s/all-namespaces/networkpolicies',
+      createResource: true,
+      selectors: [tableColumnHeaders],
+    })
+    .set('Policy details', {
+      url: '/k8s/all-namespaces/networkpolicies/example',
+      selectors: [sectionHeadings, detailLabels, i18nTranslated],
+      testIDs: ['timestamp'],
+      deleteResource: 'Network Policy',
+    });
 
-  it('pseudolocalizes timestamps', () => {
-    cy.log('test timestamps');
-    cy.visit('/k8s/all-namespaces/events?pseudolocalization=true&lng=en');
-    cy.byTestID('timestamp').isPseudoLocalized();
-  });
+  testObjs.forEach(
+    ({ url, createResource = false, deleteResource, testIDs = [], selectors = [] }, resource) => {
+      it(`pseudolocalizes ${resource}`, () => {
+        const pseudoLocUrl = `${url}?pseudolocalization=true&lng=en`;
```

Comment:
It's better to use `URLSearchParams` (or the Node equivalent if needed) to build the URL so we correctly handle URLs that already have params.

Reply:
```suggestion
        const pseudoLocURL = `${url}?pseudolocalization=true&lng=en`;
```

Reply:
fixed